modules/blast.nf

```diff
@@ -105,9 +142,46 @@ process filter_best_bitscore {
 
     output:
     tuple val(sample_id), path("${sample_id}_${db_id}_blast_best_bitscore.csv"), emit: blast_best_bitscore_csv
+    tuple val(sample_id), path("${sample_id}_${db_id}_filter_bitscore_provenance.yml"),            emit: provenance
 
     script:
     """
     filter_best_bitscore.py -i ${full_blast_report} > ${sample_id}_${db_id}_blast_best_bitscore.csv
+
+    cat <<-EOL_VERSIONS > ${sample_id}_${db_id}_filter_bitscore_provenance.yml
+    - process_name: "${task.process}-${db_id}"
```

Comment:
`process_name` should match what gets reported in the nextflow trace and nextflow report as closely as possible. I'm not sure if that should include the `tag` or not.

Reply:
I was torn on this one, because there are 3 processes all called "blastn" but run on separate databases.  I wanted the ability to distinguish them in the final provenance file. Does your schema permit duplicate process names? (i.e. 3 process entries all called "blastn").

Reply:
There's no restriction on having duplicate process names. So I'd say it would be ok to have multiple instances of a `blastn` process in a provenance file, with different database info contained in the process provenance.

Reply:
I reverted the `process_name` fields to match the exact process name in the nextflow trace. I also opted to include your tentative optional `databases` list. Here is the final result:

```yaml
- process_name: "blastn"
  tools:
  - tool_name: blastn
    tool_version: 2.13.0+
    parameters:
    - parameter: "perc_identity"
      value: 95.0
    - parameter: "qcov_hsp_perc"
      value: 95.0
  - tool_name: taxonkit
    tool_version: v0.14.1
  - tool_name: python
    tool_version: 3.11.6
  databases:
  - database_name: 16s-ezbio
    database_version: 1.0
    files: 
    - filename: /sharehome/john.palmer/work/16s/db/ezbio/2023-12-12_1.0_16s-ezbio/16s-ezbio
      sha256: b3d850bd8b1c5faeeada505c03a5a01e132de3a1abac944832b9de652eaf1122
 ```